src/main/java/com/duckblade/osrs/toa/features/het/solver/HetSolver.java

```diff
@@ -84,15 +85,24 @@ public void onGameStateChanged(GameStateChanged e)
 	@Subscribe
 	public void onChatMessage(ChatMessage e)
 	{
-		if (e.getMessage().startsWith("Your party failed to complete the challenge"))
+		if (e.getMessage().startsWith("Your party failed to complete the challenge") || e.getMessage().endsWith("violently!"))
 		{
 			clearSolve();
 		}
+		else if (e.getMessage().startsWith("Challenge complete"))
+		{
+			shutDown();
+		}
 	}
 
 	@Subscribe
 	public void onGameObjectSpawned(GameObjectSpawned e)
 	{
+		if (solution == null && e.getGameObject().getId() == NullObjectID.NULL_29733)
+		{
+			return;
+		}
```

Comment:
What is this part doing?

Reply:
From what I see it reevaluates the solution every time a gameobject is spawned. The null object is spawned after the seal breaks which triggers resolving the solution and is why the overlay is persisting.